.pre-commit-config.yaml

```
@@ -7,45 +7,45 @@ repos:
     hooks:
       - id: black
         language_version: python3
-        args: ['--target-version', 'py37']
-  - repo: https://github.com/timothycrosley/isort
-    # isort config is in setup.cfg
-    rev: 5.6.4
+        args: ['--target-version', 'py38']
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: 'v0.0.290'
     hooks:
-      - id: isort
-  - repo: https://github.com/pycqa/flake8
-    # flake8 config is in setup.cfg
-    rev: 3.8.4
-    hooks:
-      - id: flake8
-        additional_dependencies:
-          - flake8-comprehensions
-          - flake8-assertive
-          - flake8-print
-  - repo: https://github.com/rtts/djhtml
-    rev: v1.4.13
-    hooks:
-      - id: djhtml
+      - id: ruff
+        args: [--fix, --exit-non-zero-on-fix]
+
   - repo: https://github.com/pre-commit/mirrors-prettier
-    rev: v2.5.1
+    rev: v3.0.3
     hooks:
       - id: prettier
-        types_or: [css, javascript, json, yaml]
+        types_or: [css, scss, javascript, ts, tsx, json, yaml]
+        additional_dependencies:
+          # Keep in sync with package.json
+          - prettier@3.0.1
```

```suggestion
        types_or: [css, javascript, json, yaml]
```

I don’t think we want to sync this with Wagtail to _that_ extent. First off `additional_dependencies` is useless if we install Prettier only, and for this project we wouldn’t currently use SCSS / TypeScript.

---

I was going too fast and generally being a copy/paste fool. I did mean to circle back and add `ruff format` and upgrade ruff now that Wagtail is doing the same. Another PR for another day.